dataset/data_loader/SCAMPSLoader.py

```
@@ -54,6 +54,20 @@ def get_data(self, data_path):
             dirs.append({"index": subject, "path": data_dir})
         return dirs
 
+    def get_data_subset(self, data_dirs, begin, end):
+        """Returns a subset of data dirs, split with begin and end values"""
+        if begin == 0 and end == 1: # return the full directory if begin == 0 and end == 1
+            return data_dirs
+
+        file_num = len(data_dirs)
+        choose_range = range(int(begin * file_num), int(end * file_num))
+        data_dirs_new = []
+
+        for i in choose_range:
+            data_dirs_new.append(data_dirs[i])
+
+        return data_dirs_new
+
```

Same here as well. But I think we can just make sure the users use 0 and 1.0 as BEGIN and END. Otherwise, we can raise an error.

---

Same respose as UBFC. The system isolates raw data for each file split and the gathers ALL the corresponding processed files.